runtime/compiler/optimizer/DataAccessAccelerator.cpp

```diff
@@ -90,6 +90,21 @@ int32_t TR_DataAccessAccelerator::perform()
    {
    int32_t result = 0;
 
+   /* Disable DAA if vector packed decimal facilities aren't available to avoid
+    *
+    * Disabling the DAA optimization for non-vectorized paths temporarily to unblock
+    * off-heap enablement while we continue to investigate the bug in non-vectorized
+    * pdload evaluator.
+    */
+   if (comp()->target().cpu.isZ() && TR::Compiler->om.isOffHeapAllocationEnabled() &&
+      !(comp()->target().cpu.supportsFeature(OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL) &&
```

Comment:
@VermaSh Vector Packed decimal facility was introduced in z14 and in later version we added enhancement to that. Do you think we can have `OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL_ENHANCEMENT_FACILITY` or other enhancements without the basic `OMR_FEATURE_S390_VECTOR_PACKED_DECIMAL` ?

I would even look into the way we enable the vector BCD generation (Which is done if we are on z14 or newer. So may be a better query here is to check if `TR_DisableVectorBCD` option is set with OffHeap enabled. Also why can't this check added to the condition at line 108? and you need this separate block?

Reply:
I have updated the PR to check just `TR_DisableVectorBCD`. Having it separate feels more readable as compared to adding it on 108. For reference, this is what it would look like on line 108:
```
if (...
    ((TR::Compiler->om.isOffHeapAllocationEnabled() && cg->comp()->getOption(TR_DisableVectorBCD))
         || !TR::Compiler->om.isOffHeapAllocationEnabled()))
```
If you prefer I can move it to 108.